src/main/webapp/app/exercises/programming/manage/grading/feedback-analysis/feedback-analysis.component.ts

```diff
@@ -1,34 +1,170 @@
+        }
+    }
+
+    setPage(newPage: number): void {
+        this.page.set(newPage);
+        this.loadData();
+    }
+
+    async search(searchTerm: string): Promise<void> {
+        this.page.set(1);
+        this.searchTerm.set(searchTerm);
+        this.debounceLoadData();
+    }
+
+    openFeedbackModal(feedbackDetail: FeedbackDetail): void {
+        const modalRef = this.modalService.open(FeedbackModalComponent, { centered: true });
+        modalRef.componentInstance.feedbackDetail = signal(feedbackDetail);
+    }
+
+    setSortedColumn(column: string): void {
+        if (this.sortedColumn() === column) {
+            this.sortingOrder.set(this.sortingOrder() === SortingOrder.ASCENDING ? SortingOrder.DESCENDING : SortingOrder.ASCENDING);
+        } else {
+            this.sortedColumn.set(column);
+            this.sortingOrder.set(SortingOrder.ASCENDING);
+        }
+        this.loadData();
+    }
+
+    async openFilterModal(): Promise<void> {
+        const savedTasks = this.localStorage.retrieve(this.FILTER_TASKS_KEY);
+        const savedTestCases = this.localStorage.retrieve(this.FILTER_TEST_CASES_KEY);
+        const savedOccurrence = this.localStorage.retrieve(this.FILTER_OCCURRENCE_KEY);
+
+        this.minCount.set(0);
+        this.maxCount.set(await this.feedbackAnalysisService.getMaxCount(this.exerciseId()));
+        const modalRef = this.modalService.open(FeedbackFilterModalComponent, { centered: true, size: 'lg' });
+
+        modalRef.componentInstance.filterForm.setValue({
+            tasks: this.hasAppliedFilters ? savedTasks : [],
+            testCases: this.hasAppliedFilters ? savedTestCases : [],
+            occurrence: this.hasAppliedFilters ? savedOccurrence : [this.minCount(), this.maxCount()],
+        });
+
+        modalRef.componentInstance.totalAmountOfTasks = this.totalAmountOfTasks;
+        modalRef.componentInstance.testCaseNames = this.testCaseNames;
+        modalRef.componentInstance.exerciseId = this.exerciseId;
+        modalRef.componentInstance.maxCount = this.maxCount;
+        modalRef.componentInstance.filterApplied.subscribe((filters: any) => {
+            this.applyFilters(filters);
+        });
+    }
+
+    applyFilters(filters: any): void {
+        this.selectedFiltersCount.set(this.countAppliedFilters(filters));
+        this.hasAppliedFilters = this.selectedFiltersCount() !== 0;
+        this.loadData();
+    }
+
+    countAppliedFilters(filters: any): number {
+        let count = 0;
+        if (filters.tasks && filters.tasks.length > 0) {
+            count += filters.tasks.length;
+        }
+        if (filters.testCases && filters.testCases.length > 0) {
+            count += filters.testCases.length;
+        }
+        if (filters.occurrence && (filters.occurrence[0] !== 0 || filters.occurrence[1] !== this.maxCount())) {
+            count++;
         }
+        return count;
     }
```

Comment:
Using `any` could be risky here. If we change the way filters are structured, we might not find out until a runtime error happens. Would it be possible to define a type that has this information?